src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -1393,7 +1393,7 @@ public void updateMovement() {
     }
 
     public void addMovement(double x, double y, double z, double yaw, double pitch, double headYaw) {
-        this.level.addEntityMovement(this, x, y, z, yaw, pitch, headYaw);
+        this.level.addEntityMovement(this, x, y + this.getBaseOffset(), z, yaw, pitch, headYaw);
```

Comment:
I don't think you should change the use of the base method since it would break the plugins.

Reply:
The main issue stems from EntityHuman and Player both overriding addMovement() to instead send MovePlayerPackets, where they add in the getEyeHeight(), but also calling to updateMovement, which is not overriden from Entity class. 

This shouldn't break any plugins that are calling updateMovement() in their onUpdates or entityBaseTick. 

The whole movement processing for players vs. entities (and then even different for EntityHumans that are not Players!) is a little out of hand to be honest.